site_scons/buildutils.py

```diff
@@ -34,6 +36,409 @@
     TPathLike = TypeVar("TPathLike", Path, str)
 
 
+class Option:
+    """Object corresponding to SCons configuration option.
+
+    The purpose of this class is to collect system-dependent default parameters for
+    SCons configuration options, which are made available to both help text and
+    reST documentation. The class works in conjunction with the `Configuration` class
+    to select and convert system-dependent parameters to SCons configuration options.
+    """
+
+    def __init__(self,
+            name: str,
+            description: str,
+            default: "Union[bool, str]",
+            choices: list = []):
```

Comment:
Hm. Ended up turning on `"python.analysis.typeCheckingMode": "basic"` as this certainly goes way beyond what I've been doing elsewhere. For whatever reason `choices: Iterator[str] = ()` triggers:
```
Expression of type "tuple[()]" cannot be assigned to parameter of type "Iterator[str]"
  "tuple[()]" is incompatible with protocol "Iterator[str]"
    "__next__" is not presentPylancereportGeneralTypeIssues)
```

Reply:
Maybe try `Sequence` instead?

Reply:
Ugh, this is an open issue in the `typing` repo: https://github.com/python/typing/issues/256 I also suppose that `Sequence/Iterable[str] = ()` should fail, because that's an empty sequence, not a sequence of `str`s. Blergh, maybe the `None` sentinel is the best way to go. It avoids the mutable default argument problem, for sure :joy:

Reply:
Well, on close inspection, the `__init__` declaration actually has a few more items:
```Python
    def __init__(self,
            name: str,
            description: str,
            default: "Union[str, bool, Dict[str, Union[str, bool]]]",
            choices: "Union[Optional[Iterator[str]], SCVariables]" = None):
```